### PR TITLE
chore: dynamically generate GENERATED_SERVICES var

### LINKED ISSUE #861: Split Makefile `GENERATED_SERVICES` into one line per service to reduce the number of merge conflicts
### What problem are you facing?
We have to rebase for every new service using code gen when two PRs are in flight at the same time. Additionally, some services like ec2 has been missing from the Makefile.

### How could Crossplane help solve your problem?
We could reduce the likelyhood of merge conflict by:
#### A) Removing `GENERATED_SERVICES` and use find or similar to generate the list
```
find . -name generator-config.yaml | cut -d/ -f 3
apigatewayv2
dynamodb
ec2
efs
kms
rds
secretsmanager
sfn
cloudfront
lambda
servicediscovery
route53resolver
kafka
docdb
transfer
elasticache
```

#### B) One line per entry (using `define`?)
https://stackoverflow.com/a/649462

### PR DESCRIPTION
**Pull request by dwerder**
Signed-off-by: Daniel Werdermann <daniel.werdermann@gmail.com>

### Description of your changes

The variable in the Makefile is now generated from the paths of the generatec-config.yaml files. So no more editingof the Makefile and no merge conflicts.

Fixes #861
Fixes #862

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

`make reviewable` works without changes.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by haarchri** [comment]
good to go with excluded services - since we checked:
https://github.com/crossplane/provider-aws/pull/1003
https://github.com/crossplane/provider-aws/pull/1004
https://github.com/crossplane/provider-aws/pull/1006

**Comment by dwerder** [comment]
The force push commit added a more secure regex for the excludes.

**Comment by haarchri** [comment]
as of review from @muvaf  for https://github.com/crossplane/provider-aws/pull/862 it would be great to see his review here

File: Makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ PROJECT_REPO := github.com/crossplane/$(PROJECT_NAME)
 PLATFORMS ?= linux_amd64 linux_arm64
 
 CODE_GENERATOR_COMMIT ?= 285d87b66b62fbfb859986ddf74c9f9b6ae743fb
-GENERATED_SERVICES="apigatewayv2,athena,cloudfront,cloudwatchlogs,dynamodb,elbv2,ec2,efs,glue,iot,kafka,kms,lambda,mq,rds,secretsmanager,servicediscovery,sfn,transfer,ram"
+GENERATED_SERVICES := $(shell find ./apis/ -type f -name generator-config.yaml -regextype posix-extended  -not -regex  ".*/(eks|docdb|route53resolver)/.*" | cut -d/ -f 3 | tr '\n' ' ')
```
Comment (muvaf) [inline Makefile:10]: Why do we need to skip those services?

File: CODE_GENERATION.md [hunk 2]
```diff
@@ -54,9 +54,11 @@ If `apis/<serviceid>` is created from scratch, please add the new service name t
 
 ### Makefile
 
+The serviceids are fetch from the paths of the generator-config.yaml automatically. If you want explicitly exclude a service adjust the Makefile:
```
Comment (muvaf) [inline CODE_GENERATION.md:57]: How common is the exclusion need?

File: Makefile (hunk 1 above)
Comment (haarchri) [reply Makefile:10]: today we missing eks, route53resolver and docdb in makefile - 
 check:
 https://github.com/crossplane/provider-aws/pull/1006
https://github.com/crossplane/provider-aws/pull/1004
https://github.com/crossplane/provider-aws/pull/1003

Comment (haarchri) [reply Makefile:10]: #1006 #1004 #1003 fixed

File: CODE_GENERATION.md [hunk 3]
```diff
@@ -55,21 +55,18 @@ resources:
...
 ### Makefile
 
-```bash
-- GENERATED_SERVICES="apigatewayv2"
-+ GENERATED_SERVICES="apigatewayv2,<serviceid>"
-```
+The serviceids are fetch from the paths of the generator-config.yaml automatically.
 
 If this step fails for some reason, please raise an issue in [code-generator](https://github.com/aws-controllers-k8s/code-generator)
 and mention that you're using Crossplane pipeline.
```
Comment (muvaf) [inline CODE_GENERATION.md:72]: Let's remove the whole section as they don't need to know about this at all thanks to your discovery implementation.

**Review by muvaf — APPROVED** [review]
Thanks @dwerder ! LGTM with only one thing to address. Great work!

File: CODE_GENERATION.md (hunk 3 above)
Comment (dwerder) [reply CODE_GENERATION.md]: Done

**Review by muvaf — APPROVED** [review]

**Pull request merged**